pkg/operation/botanist/secrets.go

```diff
@@ -546,32 +547,64 @@ func (b *Botanist) DeleteOldServiceAccountSecrets(ctx context.Context) error {
 // RewriteSecretsAddLabel patches all secrets in all namespaces in the shoot clusters and adds a label whose value is
 // the name of the current ETCD encryption key secret. This function is useful for the ETCD encryption key secret
 // rotation which requires all secrets to be rewritten to ETCD so that they become encrypted with the new key.
+// After it's done, it snapshots ETCD so that we can restore backups in case we lose the cluster before the next
+// incremental snapshot is taken.
 func (b *Botanist) RewriteSecretsAddLabel(ctx context.Context) error {
```

Comment:
The name of the method does not suggest that there is a ETCD backup taken here and it is not mentioned [in its task name](https://github.com/rfranzke/gardener/blob/652efceb30439359fda8eb59cd000695a76db7bd/pkg/gardenlet/controller/shoot/shoot_control_reconcile.go#L360-L361) yet

Reply:
I moved it now into a separate function